packages/calypso-products/src/utils.js

```diff
@@ -2,7 +2,7 @@
  * External dependencies
  */
 
-import { map, head, property } from 'lodash';
-
-export const findCurrencyFromPlans = ( plans ) =>
-	head( map( plans, property( 'currency_code' ) ) ) || 'USD';
+export const findCurrencyFromPlans = ( plans ) => {
+	const firstPlanWithCurrency = plans.find( ( plan ) => plan.currency_code );
+	return firstPlanWithCurrency?.currency_code ?? 'USD';
+};
```

Comment:
The original function does something different: it looks at the first `plan` in `plans` (never at any other ones) and returns its `plan.currency_code`. If that value is falsy, defaults to `USD`.

Reply:
Good catch.